src/index.js

```diff
@@ -17,20 +17,32 @@ export default function createDb(source: string, options: StoreOptions = {}): Fu
   if (storage) {
     dbObject = init(source, storage.read, storage.write)
   } else {
-    dbObject = map({})
+    dbObject = obsMap({})
   }
 
   autorun(function() {
     states.push(serializeDb(dbObject))
   })
 
-  function db(key: string): Array<any> {
+  function db(key: string, funcs?: Array<Function> | Function): Array<any> {
     // If the observable array doesn't exist create it
     if (!dbObject.has(key)) dbObject.set(key, createData(dbObject, key, []))
-    return dbObject.get(key).__data
+    if (funcs) {
+      const updateObs = partial(update, dbObject, key)
+      if (Array.isArray(funcs)) {
```

Comment:
You could remove an extra case here by converting the single to array using something like
```
const arrayFuncs = [].concat(funcs)
```
so both `function(){}` and `[function(){}]` would result in `[function(){}]` for you